projects/plugins/crm/admin/settings/general.page.php

```diff
@@ -342,14 +343,29 @@
 				</tr>
 
 				<tr>
-					<td class="wfieldname"><label for="wpzbscrm_secondaddress"><?php esc_html_e( 'Second Address Fields', 'zero-bs-crm' ); ?>:</label><br /><?php esc_html_e( 'Allow editing of a "second address" against a contact', 'zero-bs-crm' ); ?></td>
+					<td class="wfieldname"><label for="wpzbscrm_secondaddress"><?php esc_html_e( 'Second Address Fields Visibility', 'zero-bs-crm' ); ?>:</label><br /><?php esc_html_e( 'Choose "second address" visibility in the contact edit page. It can be hidden, visible, or there will be an option to toggle it.', 'zero-bs-crm' ); ?></td>
 					<td style="width:540px">
-						<input type="checkbox" class="winput form-control" name="wpzbscrm_secondaddress" id="wpzbscrm_secondaddress" value="1"
-						<?php
-						if ( isset( $settings['secondaddress'] ) && $settings['secondaddress'] == '1' ) {
-							echo ' checked="checked"';}
-						?>
-						/>
+						<select class="winput" name="wpzbscrm_secondaddress" id="wpzbscrm_secondaddress">
+							<option value="<?php echo esc_attr( SETTING_SECOND_ADDRESS_ALWAYS_HIDE ); ?>"
+							<?php
+							$jpcrm_setting_second_address_visibility = jpcrm_settings_get_second_address_visibility( $settings );
+							if ( $jpcrm_setting_second_address_visibility === SETTING_SECOND_ADDRESS_ALWAYS_HIDE ) {
+								echo ' selected="selected"';}
+							?>
+							><?php esc_html_e( 'Always Hide', 'zero-bs-crm' ); ?></option>
+							<option value="<?php echo esc_attr( SETTING_SECOND_ADDRESS_ALWAYS_SHOW ); ?>"
+							<?php
+							if ( $jpcrm_setting_second_address_visibility === SETTING_SECOND_ADDRESS_ALWAYS_SHOW ) {
+								echo ' selected="selected"';}
+							?>
+							><?php esc_html_e( 'Always Show', 'zero-bs-crm' ); ?></option>
+							<option value="<?php echo esc_attr( SETTING_SECOND_ADDRESS_ALLOW_CHANGE_VISIBILITY ); ?>"
+							<?php
+							if ( $jpcrm_setting_second_address_visibility === SETTING_SECOND_ADDRESS_ALLOW_CHANGE_VISIBILITY ) {
+								echo ' selected="selected"';}
+							?>
+							><?php esc_html_e( 'Manually Toggle', 'zero-bs-crm' ); ?></option>
+						</select>
```

Comment:
See how messy and unreadable this gets? "Quotes + Accepted status" is messy like this too. At minimum we should have a helper function that distills it into one setting, but I still think it should be two settings.